client/gutenberg/editor/calypsoify-iframe.jsx

```diff
@@ -29,16 +51,87 @@ const getIframeUrl = ( siteAdminUrl, postId, postType ) => {
 };
 
 class CalypsoifyIframe extends Component {
+	state = {
+		isMediaModalVisible: false,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.iframe = null;
+	}
+
+	componentDidMount() {
+		window.addEventListener( 'message', this.onMessage, false );
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener( 'message', this.onMessage, false );
+	}
+
+	setIframeRef = element => {
+		this.iframe = element;
+	};
+
+	onMessage = ( { data } ) => {
+		if ( typeof data !== 'string' || data[ 0 ] !== '{' ) {
+			return;
+		}
+		const message = JSON.parse( data );
+		const { action, type, payload } = message;
+
+		if ( type !== 'gutenbergIframeMessage' ) {
+			return;
+		}
+
+		if ( action === 'openMediaModal' ) {
+			const { gallery, multiple, allowedTypes } = payload;
+			this.setState( { isMediaModalVisible: true, gallery, multiple, allowedTypes } );
+		}
+	};
+
+	openMediaModal = () => {
+		if ( ! this.state.isMediaModalVisible ) {
+			this.setState( { isMediaModalVisible: true } );
+		}
+	};
+
+	closeMediaModal = media => {
+		if ( media ) {
+			const { multiple } = this.state;
+			const formattedMedia = map( media.items, item => mediaCalypsoToGutenberg( item ) );
+			const payload = multiple ? formattedMedia : formattedMedia[ 0 ];
+
+			sendMessage( this.iframe, {
+				action: 'selectMedia',
+				payload,
+			} );
+		}
+
+		this.setState( { isMediaModalVisible: false } );
+	};
+
 	render() {
-		const { iframeUrl } = this.props;
+		const { iframeUrl, siteId } = this.props;
+		const { isMediaModalVisible, multiple, allowedTypes } = this.state;
 
 		return (
 			<Fragment>
 				{ /* eslint-disable-next-line wpcalypso/jsx-classname-namespace */ }
 				<div className="main main-column calypsoify is-iframe" role="main">
 					{ /* eslint-disable-next-line jsx-a11y/iframe-has-title, wpcalypso/jsx-classname-namespace */ }
-					<iframe className={ 'is-iframe-loaded' } src={ iframeUrl } />
+					<iframe ref={ this.setIframeRef } className={ 'is-iframe-loaded' } src={ iframeUrl } />
```

Comment:
I'm only the messenger 😉 , but I believe that React has advised using this [third round of `ref` setting](https://reactjs.org/docs/react-api.html#reactcreateref)…

```js
constructor() {
	this.iframe = React.createRef();
}

<iframe ref={ this.iframe } />
```

Reply:
feels like this changes with every version of React... updated on a3a55ead10cd65760b02a810f2d4e197071e308a.